src/Attestation/Attestation/Commands/RemoveAzureAttestation.cs

```diff
@@ -0,0 +1,125 @@
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = NameParameterSet,
+            HelpMessage = "Specifies the name of the attestation to remove.")]
+        [ResourceNameCompleter("Microsoft.Attestation", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+
+        /// <summary>
+        /// Attestation object
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Attestation Resource Id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+
+        /// <summary>
+        /// Attestation object
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Attestation object to be deleted.")]
+        [ValidateNotNullOrEmpty]
+        public PSAttestation InputObject { get; set; }
+
+
+        /// <summary>
+        /// Resource group to which the attestation belongs.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 1,
+            ParameterSetName = NameParameterSet,
+            HelpMessage = "Specifies the name of resource group for Azure attestation to remove.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public string ResourceGroupName { get; set; }
+
+
+        /// <summary>
+        /// If present, do not ask for confirmation
+        /// </summary>
+        [Parameter(Mandatory = false,
+        HelpMessage = "Indicates that the cmdlet does not prompt you for confirmation. By default, this cmdlet prompts you to confirm that you want to delete the attestation.")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "This Cmdlet does not return an object by default. If this switch is specified, it returns true if successful.")]
+        public SwitchParameter PassThru { get; set; }
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            if (InputObject != null)
+            {
+                Name = InputObject.Name;
+                ResourceGroupName = InputObject.ResourceGroupName;
+            }
+            else if (ResourceId != null)
+            {
+                var resourceIdentifier = new ResourceIdentifier(ResourceId);
+                Name = resourceIdentifier.ResourceName;
+                ResourceGroupName = resourceIdentifier.ResourceGroupName;
+            }
+
+            AttestationClient.DeleteAttestation(Name,ResourceGroupName);
```

Comment:
@shleiAmy please make sure this call is wrapped in the `ShouldProcess` check

Reply:
fixed